plugins/FreeBoy/GbApuWrapper.h

```diff
@@ -31,25 +32,30 @@ namespace lmms
 {
 
 
-class Gb_Apu_Buffer : public Gb_Apu {
+class GbApuWrapper : private Gb_Apu
+{
 	MM_OPERATORS
 public:
-	Gb_Apu_Buffer() = default;
-	~Gb_Apu_Buffer() = default;
+	GbApuWrapper() = default;
+	~GbApuWrapper() = default;
 
-	void end_frame(blip_time_t);
+	blargg_err_t setSampleRate(long sampleRate, long clockRate);
+	void writeRegister(unsigned addr, int data) { Gb_Apu::write_register(fakeClock(), addr, data); }
+	long samplesAvail() const;
+	long readSamples(blip_sample_t* out, long count);
+	void trebleEq(const blip_eq_t& eq) { Gb_Apu::treble_eq(eq); }
+	void bassFreq(int freq);
+	void endFrame(blip_time_t endTime);
```

Comment:
I'm not sure if it's better to match the original code style of the submoduled upstream library. @tresf Any ideas, since you wrote this class?

Reply:
Even with upstream changes, we'd still need a wrapper class for handling the CPU time. I don't think it's really worth it or should be within the scope of this simple bug fix PR. But after this PR is complete, I can look into upgrading to the latest version of the library and making significant structural changes like that.

Reply:
Thanks... Yeah, my intentions aren't to delay anything.  I'm no longer an active contributor to LMMS, so I'm mostly brainstorming here. The code style guidelines are a wider project decision and since the codebase has recently received a huge effort to standardize all of the codebase, it's only natural that these types of exceptions need to be decided upon.

@JohannesLorenz / @PhysSong if the decision is to maintain the submodile style, is there a quick way make this process easier (or at least more straight forward) for contributors?

For example, the switch to the submodule deliberately changed the plugin name to match our internal conventions, however some wrapper code just doesn't benefit from forced styling due to divergence from upstream.

The project should have guidelines for this so that contributions aren't being held up by missing policy.

Reply:
I would say: our files -> our style. Their files -> their style. If we start changing style in their submodule, we risk ugly merge errors when we update from their upstream. If we use their style in our files, we risk issues with our code check/format tools and also difficulties amongst our developers. Of course, this can mean that some of our files use mixed styles in the same source file, but we have this already with STL or Qt. Though, it's just my opinion and I am open to other suggestions.